nservicebus/hosting/service-fabric-hosting/index.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Service Fabric Hosting
+related:
+ - nservicebus/service-fabric
+ - samples/azure/azure-service-fabric-routing
+reviewed: 2017-03-30
+---
+
+NServiceBus endpoints can be hosted in Service Fabric using any of these three options:
+
+1. Stateless services
+1. Stateful services
+1. Guest executable 
+
+To decide what option to use, refer to [Service Fabric documentation](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-overview).
+
+Note: The Actor model is another Service Fabric programming model that is currently not supported by NServiceBus.
+
+### Stateless service
+
+Hosting with a Stateless service is very similar to any other Azure-based hosting (using [Cloud services](/nservicebus/hosting/cloud-services-host) or [self-hosting](/nservicebus/hosting/#self-hosting) with [Azure App Service](https://docs.microsoft.com/en-us/azure/app-service/)). Endpoints are stateless and use storage external to Service Fabric for managing data needed for their operation. Endpoints can be scaled out, leveraging competing consumer at the transport level.
+
+With stateless services, the number of instances of a service can range from one to the number of nodes in a cluster (-1). Endpoints are self hosted and should be started using a custom Service Fabric [`ICommunicationListener`](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-communication) implementation.
+
+snippet: StatelessEndpointCommunicationListener
+
+### Stateful service
+
+The process of configuring and starting an endpoint hosted in a stateful service is very similar to stateless service hosting approach. The main difference is that the endpoint cannot be started until after the `OpenAsync` method has returned as the `StateManager` instance has not been fully configured yet. This is fairly trivial to solve by calling the `Endpoint.Start` operation from the `RunAsync` operation instead.
+
+snippet: StatefulEndpointCommunicationListener
+
+snippet: StatefulService
+
+Due to characteristics of Stateful Services i.e. data partitioning and local data storage, additional configuration aspects have to be taken into consideration: 
+
+- Service partitioning schema must be well defined before the first deployment occurs (repartitioning requires all data to be deleted)
+- Messages must be routed among the shards according to the partitioning schema
+- There can only be one NServiceBus endpoint running per service partition (in the master instance)
```

Comment:
Personally, I find "Endpoint instances can be running only on primary replicas" confusing as well. Bottom line that we want to say is that "there will be only one active instance of an endpoint per partition".